virtualizarr/tests/test_readers/test_hdf/test_hdf.py

```diff
@@ -16,19 +16,18 @@ class TestDatasetChunkManifest:
     def test_empty_chunks(self, empty_chunks_hdf5_file):
         f = h5py.File(empty_chunks_hdf5_file)
         ds = f["data"]
-        with pytest.raises(ValueError, match="chunked but contains no chunks"):
-            HDFVirtualBackend._dataset_chunk_manifest(
-                path=empty_chunks_hdf5_file, dataset=ds
-            )
+        manifest = HDFVirtualBackend._dataset_chunk_manifest(
+            path=empty_chunks_hdf5_file, dataset=ds
+        )
+        assert manifest.shape_chunk_grid == (0,)
 
-    @pytest.mark.skip("Need to differentiate non coordinate dimensions from empty")
     def test_empty_dataset(self, empty_dataset_hdf5_file):
         f = h5py.File(empty_dataset_hdf5_file)
         ds = f["data"]
-        with pytest.raises(ValueError, match="no space allocated in the file"):
-            HDFVirtualBackend._dataset_chunk_manifest(
-                path=empty_dataset_hdf5_file, dataset=ds
-            )
+        manifest = HDFVirtualBackend._dataset_chunk_manifest(
+            path=empty_dataset_hdf5_file, dataset=ds
+        )
```

Comment:
Generally it's better to write tests that use higher-level public API instead of directly calling internals if you can help it, but that's just a nit.

Reply:
👍 I think the tests will be easier to structure when we move away from `VirtualBacked` base to something more functional and the returned `ManifestStore` will be simpler to introspect and reason about than a complete `xr.Dataset`.